contracts/contracts/rem.system/src/voting.cpp

```diff
@@ -248,8 +249,8 @@ namespace eosiosystem {
             const auto active_prod = std::find_if(std::begin(_gstate.last_schedule), std::end(_gstate.last_schedule),
                [target = pd.first](const auto& prod){ return prod.first.value == target.value; });
             if (active_prod != std::end(_gstate.last_schedule)) {
-               _gstate.total_active_producer_vote_weight += pd.second.first;
-               active_prod->second = pitr->total_votes / _gstate.total_active_producer_vote_weight;
+               _gstate.total_active_producer_vote_weight += pitr->total_votes - total_votes_before;
+               update_pervote_shares();
```

Comment:
@me-vadim can we move this out of the loop?

Reply:
Done